src/main/java/it/smartcommunitylab/aac/spid/provider/SpidIdentityProviderConfig.java

```diff
@@ -94,6 +97,51 @@ private SpidIdentityProviderConfig() {
         super();
     }
 
+    public void setBaseUrl(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    public SpidIdentityProviderStatusMap getStatusMap() {
+        if (statusMap == null) {
+
+            statusMap = new SpidIdentityProviderStatusMap();
+            statusMap.setMetadataUrl(getMetadataUrl());
+            statusMap.setAssertionConsumerUrls(getAssertionConsumerUrls());
+        }
+        return statusMap;
+    }
+
+    public String getMetadataUrl() {
+        if (baseUrl != null) {
+            UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(metadataUrlTemplate());
+            return builder.buildAndExpand(Map.of("baseUrl", baseUrl, "registrationId", getMetadataRegistrationId())).toUriString();
+        }
+        return null;
+    }
+
+    public String metadataUrlTemplate() {
+        return "{baseUrl}/auth/" + getAuthority() + "/metadata/{registrationId}";
+    }
+
+    public Map<String, String> getAssertionConsumerUrls(){
+        if (baseUrl != null) {
+            Map<String, String> assertionConsumerUrls = new HashMap<>();
+            for(RelyingPartyRegistration relyingPartyRegistration : getUpstreamRelyingPartyRegistrations()){
+                if(relyingPartyRegistration.getProviderDetails().getEntityId() != null) {
+                    assertionConsumerUrls.put(
+                            relyingPartyRegistration.getProviderDetails().getEntityId(),
+                            UriComponentsBuilder.fromUriString(identityProviderUrlTemplate()).buildAndExpand(Map.of("baseUrl", baseUrl, "registrationId", relyingPartyRegistration.getRegistrationId())).toUriString());
+                    }
+                }
+            return assertionConsumerUrls;
+        }
+        return null;
+    }
+
+    public String identityProviderUrlTemplate() {
```

Comment:
keep this method consistent: this produces an `assertionConsumerUrlTemplate`, as used in `getAssertionConsumerUrls`.